packages/next-auth/src/next/index.ts

```diff
@@ -1,63 +1,50 @@
-import { NextAuthHandler } from "../core"
+import { AuthHandler } from "../core"
 import { detectHost } from "../utils/detect-host"
-import { setCookie } from "./utils"
+import { getBody } from "./utils"
 
 import type {
   GetServerSidePropsContext,
   NextApiRequest,
   NextApiResponse,
 } from "next"
 import type { NextAuthOptions, Session } from ".."
-import type {
-  NextAuthAction,
-  NextAuthRequest,
-  NextAuthResponse,
-} from "../core/types"
+import type { NextAuthRequest, NextAuthResponse } from "../core/types"
 
-async function NextAuthNextHandler(
+async function NextAuthHandler(
   req: NextApiRequest,
   res: NextApiResponse,
   options: NextAuthOptions
 ) {
-  const { nextauth, ...query } = req.query
-
-  options.secret =
-    options.secret ?? options.jwt?.secret ?? process.env.NEXTAUTH_SECRET
-
-  const handler = await NextAuthHandler({
-    req: {
-      host: detectHost(req.headers["x-forwarded-host"]),
-      body: req.body,
-      query,
-      cookies: req.cookies,
-      headers: req.headers,
+  options.secret ??= options.jwt?.secret ?? process.env.NEXTAUTH_SECRET
```

Comment:
We can't/shouldn't, in case a runtime could not handle `process.env` (technically it's Node.js only). `options.jwt.secret` is deprecated by a TS command already and should be removed in the future anyway.